app/app.go

```diff
@@ -75,7 +81,16 @@ func addCachableHandler(e *gin.Engine, method string, endpoint string, generator
 // / This function will act as the handler for
 // / the home page
 func homeHandler(c *gin.Context, settings common.AppSettings, db database.Database) ([]byte, error) {
-	posts, err := db.GetPosts()
+	pageNumQuery := c.Param("num")
+	pageNum, err := strconv.Atoi(pageNumQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	limit := 10 // or whatever limit you want
+	offset := (pageNum - 1) * limit
```

Comment:
If `pageNum` is 0, then this is not gonna be a nice offset! Try something like taking the max of this and zero.